syntax/gleam.vim

```diff
@@ -8,32 +8,65 @@ let b:current_syntax = "gleam"
 
 " Keywords
 syntax keyword gleamKeyword
-  \ module import pub panic use
-  \ type let as if else todo const
-  \ case assert tuple try opaque
+  \ import pub panic use type let as if else todo const case assert try opaque
 highlight link gleamKeyword Keyword
 
 " Function definition
 syntax keyword gleamDef fn nextgroup=gleamFunctionDef skipwhite skipempty
 highlight link gleamDef Keyword
 
-syntax match gleamFunctionDef "[a-z_-][0-9a-z_-]*" contained skipwhite skipnl
+syntax match gleamFunctionDef "[a-z][a-z0-9_]*\ze\s*(" skipwhite skipnl
+highlight link gleamFunctionDef Function
 highlight link gleamFunctionDef Function
 
-" Int
-syntax match gleamInt '\<\(0x[a-fA-F0-9_]\+\|[0-9][0-9_]*\)\>'
+" Number
+"" Int
+syntax match gleamInt '\<\d[0-9_]*\>'
 highlight link gleamInt Number
 
-" Float
-syntax match gleamFloat '\<[0-9][0-9_]*\.[0-9_]*\>'
+"" Binary
+syntax match gleamBinary '\<0[bB][01]+\>'
+highlight link gleamBinary Number
+
+"" Octet
+syntax match gleamOctet '\<0[oO][0-7]+\>'
+highlight link gleamOctet Number
+
+"" Hexadecimal
+syntax match gleamHexa '\<0[xX]\x+\>'
+highlight link gleamHexa Number
+
+"" Float
+syntax match gleamFloat '\d\(_\=\d\)*\.\(\d\(_\=\d\)*\)\=\(e-\=\d\)\=[0-9_]*'
```

Comment:
What does this do?

Reply:
Based on your other reviews, I changed this regex to be more precise, so I will explain the changed version.
```vim
'\(0*[1-9][0-9_]*\|0\)\.\(0*[1-9][0-9_]*\|0\)\(e-\=0*[1-9][0-9_]*\)\='


\(0*[1-9][0-9_]*\|0\)
" match zero-to-many '0' followed by a non-zero digit and then zero-to-many digit+underscore
" this is the same regex to match integers, letting us do : 00000000_113_3, which is an allowed value by gleam, so it should be highlighted. 

\.
" match a literal dot

\(0*[1-9][0-9_]*\|0\) 
" same digit matching as above

\(e-\=0*[1-9][0-9_]*\)\=
" exponant matching
" the exponant must start with an 'e' and can optionally (\= is zero-or-one in vim patterns) be followed by a minus sign.
" we then make the same digit matching than before, just leaving the zero part out, since e0 or e-0 are not valid exponant.
```

Reply:
Thank you!